apps/src/templates/instructions/BackgroundMusicMuteButton.jsx

```diff
@@ -0,0 +1,134 @@
+import i18n from '@cdo/locale';
+import {connect} from 'react-redux';
+import {PaneButton} from '@cdo/apps/templates/PaneHeader';
+import firehoseClient from '@cdo/apps/lib/util/firehose';
+import cookies from 'js-cookie';
+import {setMuteMusic, SignInState} from '@cdo/apps/templates/currentUserRedux';
+import UserPreferences from '../../lib/util/UserPreferences';
+
+const MUTE_MUSIC = 'mute_music';
+
+function BackgroundMusicMuteButton({
+  className,
+  signedIn,
+  isMinecraft,
+  isRtl,
+  currentUserBackgroundMusicMuted,
+  setMuteMusic,
+  muteBackgroundMusic,
+  unmuteBackgroundMusic
+}) {
+  const initialMuteState = signedIn
+    ? currentUserBackgroundMusicMuted
+    : !!cookies.get(MUTE_MUSIC);
+
+  const [isBackgroundMusicMuted, setIsBackgroundMusicMuted] = useState(
+    initialMuteState
+  );
+
+  const updateMuteMusic = updatedMuteValue => {
+    signedIn ? new UserPreferences().setMuteMusic(updatedMuteValue) : {};
+    setMuteMusic(updatedMuteValue);
+  };
+
+  const handleMuteMusicTabClick = () => {
+    const updatedMuteValue = !isBackgroundMusicMuted;
+
+    updateMuteMusic(updatedMuteValue);
+    setIsBackgroundMusicMuted(updatedMuteValue);
+
+    /*
+    Depending on the updated value:
+      Stop or start the background music immediately
+      Set or remove the mute cookie
+      Determine the firehose label
+    */
+    var muteLabel;
+    if (updatedMuteValue) {
+      muteBackgroundMusic();
+      cookies.set(MUTE_MUSIC, 'true', {expires: 30, path: '/'});
+      muteLabel = 'mute';
+    } else {
+      unmuteBackgroundMusic();
+      cookies.remove(MUTE_MUSIC, {path: '/'});
+      muteLabel = 'unmute';
+    }
+
+    const labType = isMinecraft ? 'Minecraft' : 'Starwars';
+
+    const record = {
+      study: 'mute-music',
+      event: 'mute-toggle',
+      data_json: JSON.stringify({
+        labType: labType,
+        muteLabel: muteLabel
+      })
+    };
+    firehoseClient.putRecord(record);
+  };
+
+  return (
+    <PaneButton
+      id={className}
+      headerHasFocus={true}
+      iconClass={isBackgroundMusicMuted ? 'fa fa-volume-off' : 'fa fa-music'}
+      label={
+        isBackgroundMusicMuted
+          ? i18n.backgroundMusicOff()
+          : i18n.backgroundMusicOn()
+      }
+      isRtl={isRtl}
+      isMinecraft={isMinecraft}
+      onClick={handleMuteMusicTabClick}
+      style={{
+        ...(!isMinecraft
+          ? isBackgroundMusicMuted
+            ? styles.musicOff
+            : styles.musicOn
+          : {})
+      }}
+    />
+  );
+}
+
+BackgroundMusicMuteButton.propTypes = {
+  className: PropTypes.string.isRequired,
+  signedIn: PropTypes.bool.isRequired,
+  isMinecraft: PropTypes.bool.isRequired,
+  isRtl: PropTypes.bool.isRequired,
+
+  // from redux
+  setMuteMusic: PropTypes.func.isRequired,
+  currentUserBackgroundMusicMuted: PropTypes.bool.isRequired,
+  muteBackgroundMusic: PropTypes.func.isRequired,
+  unmuteBackgroundMusic: PropTypes.func.isRequired
```

Comment:
To summarize some of the conversation the happened elsewhere/offline: we are moving forward with using redux because it matches the pattern on this page for other components, and fits with the parent TopInstructions component.